sklearn/utils/estimator_checks.py

```diff
@@ -3933,12 +3933,37 @@ def check_estimator_get_tags_default_keys(name, estimator_orig):
 
 
 def check_estimator_tags_renamed(name, estimator_orig):
-    assert not hasattr(estimator_orig, "_more_tags"), (
-        "_more_tags() was removed in 1.6. " "Please use __sklearn_tags__ instead.",
-    )
-    assert not hasattr(estimator_orig, "_get_tags"), (
-        "_get_tags() was removed in 1.6. " "Please use __sklearn_tags__ instead."
+    help = """_more_tags() was removed in 1.6. Please use __sklearn_tags__ instead.
+Use the following template to migrate if you wish to support multiple versions of
+scikit-learn:
+
+import sklearn
+from packaging import version
+from sklearn.base import BaseEstimator
+from sklearn.utils.metaestimators import available_if
+
+def check_version(estimator):
+    return version.parse(sklearn.__version__) < version.parse("1.6.dev")
```

Comment:
I think that we can get the `base` version to avoid to have to deal with the `.dev`

Reply:
I think it could be nice to have that in the user guide as well for the transition.

Reply:
I'm reluctant to put this in the docs since it's about private API. I think this error message in reality is where users would see it, rather than the docs.

And as for the version, the issue is that `1.5.2 < 1.6.dev < 1.6` and the change really happens in `1.6.dev`. If we use any other version, we might need to change it. I've also seen code in other repos where they reference some unreleased version of python.

Reply:
I would guess the issue is that users would define `_more_tags` (but not `__sklearn_tags__`), but not have things work as expected once they transition to v1.6?

Reply:
I think it's fine to have both methods. So, if the user defines both `_more_tags` and `__sklearn_tags__` they most probably know what they're doing.

I've changed the PR to reflect that.

Reply:
If they can safely define both, then can we remove the need for `available_if`? The story becomes "If you want to support multiple scikit-learn versions, define both."